### PR TITLE
feat: Auto-save addById / removeById

### PR DESCRIPTION
**Pull request by y-lohse**
This is a breaking change as we are removing deprecated APIs.

### CODE REVIEW

**Comment by y-lohse** [comment]
[(0.5) Contacts : cozy-client & relationships](https://trello.com/c/jv67vA73/1591-05-contacts-cozy-client-relationships)

**Comment by CozyKourai** [comment]
[(1) Choisir une instance de test dans travis](https://trello.com/c/Q1tm2Lut/1615-1-choisir-une-instance-de-test-dans-travis)

File: packages/cozy-client/src/__tests__/associations.spec.js [hunk 1]
```diff
@@ -86,29 +86,5 @@ describe('Associations', () => {
...
-
-    it('should be able to dereference docs', async () => {
-      requestHandler.mockReturnValueOnce(Promise.resolve(undefined))
-      await getTodo(TODO_1._id).attachments.remove([
-        { _id: 'abc', _type: 'io.cozy.files', name: 'abc.pdf' }
-      ])
-      expect(getTodo(TODO_1._id).attachments.data).toEqual([
-        { _id: 'def', _type: 'io.cozy.files', name: 'def.png' }
-      ])
-      expect(getTodo(TODO_1._id).attachments.count).toBe(3)
```
Comment (Crash--) [inline packages/cozy-client/src/__tests__/associations.spec.js]: Why did you remove them?

Comment (y-lohse) [reply packages/cozy-client/src/__tests__/associations.spec.js]: They were testing `HasManyFiles.add` and `HasManyFiles.remove` which no longer exist. The replacements `addById` and `removeById` [have their own tests](https://github.com/cozy/cozy-client/blob/master/packages/cozy-client/src/associations/HasManyFiles.spec.js#L37).
